GVFS/GVFS.Platform.Windows/ActiveEnumeration.cs

```diff
@@ -6,10 +6,11 @@ namespace GVFS.Platform.Windows
 {
     public class ActiveEnumeration

```

Comment:
@wilbaker is this class reset if the enumeration is restarted with a new filter?

Reply:
@CAllredMSFT Yep when the enumeration is reset the callback calls `RestartEnumeration`

```
/// <summary>
/// Resets the enumerator and advances it to the first ProjectedFileInfo in the enumeration
/// </summary>
/// <param name="filter">Filter string to save.  Can be null.</param>
public void RestartEnumeration(string filter)
{
    this.ResetEnumerator();
    this.IsCurrentValid = this.fileInfoEnumerator.MoveNext();
    this.SaveFilter(filter);
}
```

And `SaveFilter` determines which pattern matcher we use.

You can see the call to `RestartEnumeration` [here](https://github.com/Microsoft/VFSForGit/blob/355ead878bac3fade5f213aadf239332ff396224/GVFS/GVFS.Platform.Windows/WindowsFileSystemVirtualizer.cs#L498)